test/webauthn/apis.test.js

```diff
@@ -46,104 +42,730 @@ describe(`apisFunctions: ${printPath("[test/webauthn/apis.test.js]")}`, function
         await cleanST();
     });
 
-    it("test registerOptionsAPI with default values", async function () {
-        const connectionURI = await startST();
-
-        STExpress.init({
-            supertokens: {
-                connectionURI,
-            },
-            appInfo: {
-                apiDomain: "api.supertokens.io",
-                appName: "SuperTokensplm",
-                websiteDomain: "supertokens.io",
-            },
-            recipeList: [WebAuthn.init()],
+    describe("[registerOptions]", function () {
+        it("test registerOptions with default values", async function () {
+            const connectionURI = await startST();
+
+            STExpress.init({
+                supertokens: {
+                    connectionURI,
+                },
+                appInfo: {
+                    apiDomain: "api.supertokens.io",
+                    appName: "SuperTokensplm",
+                    websiteDomain: "supertokens.io",
+                },
+                recipeList: [WebAuthn.init()],
+            });
+
+            // run test if current CDI version >= 2.11
+            if (!(await isCDIVersionCompatible("2.11"))) return;
+
+            const app = express();
+            app.use(middleware());
+            app.use(errorHandler());
+
+            // passing valid field
+            let registerOptionsResponse = await new Promise((resolve) =>
+                request(app)
+                    .post("/auth/webauthn/options/register")
+                    .send({
+                        email: "test@example.com",
+                    })
+                    .expect(200)
+                    .end((err, res) => {
+                        if (err) {
+                            console.log(err);
+                            resolve(undefined);
```

Comment:
In this case we should be rejecting, resolving with undefined makes debugging tests rather confusing.